qutip/solver/brmesolve.py

```diff
@@ -274,6 +280,7 @@ def __init__(self, H, a_ops, c_ops=None, sec_cutoff=0.1, *, options=None):
         self._num_collapse = len(c_ops)
         self._num_a_ops = len(a_ops)
         self.rhs = self._prepare_rhs()
+        self._rhs = self._update_rhs()
```

Comment:
Duplicating the RHS here feels very messy. Could we just store a single RHS as `self.rhs` and convert to and from `QobjEvoHerm` as needed when options are updated?